ambry-network/src/main/java/com.github.ambry.network/BlockingChannelConnectionPool.java

```diff
@@ -216,13 +222,22 @@ public void destroyBlockingChannel(BlockingChannel blockingChannel) {
       channel.connect();
       logger.trace("Destroying connection and adding new connection for host {} port {}", host, port.getPort());
       blockingChannelAvailableConnections.add(channel);
+      currentConnectFailuresToCleanUp.set(0);
     } catch (Exception e) {
       logger
           .error("Connection failure to remote host {} and port {} when destroying and recreating the connection", host,
               port.getPort());
       synchronized (lock) {
         // decrement the number of connections to the host and port. we were not able to maintain the count
         numberOfConnections.decrementAndGet();
+        currentConnectFailuresToCleanUp.incrementAndGet();
+        // if we have reached the max count for connection failures, clean up the available connections
+        if(currentConnectFailuresToCleanUp.get() >= maxConnectFailuresToCleanUp) {
+          while(!blockingChannelAvailableConnections.isEmpty()){
+            blockingChannelAvailableConnections.poll().disconnect();
+          }
+          currentConnectFailuresToCleanUp.set(0);
+        }
```

Comment:
if you are destroying everything in the available pool, don't you need to set `numberOfConnections` accordingly?